packages/dds/tree/src/shared-tree/treeCheckout.ts

```diff
@@ -106,6 +110,31 @@ export interface CheckoutEvents {
 	 * this change is not revertible.
 	 */
 	changed(data: CommitMetadata, getRevertible?: RevertibleAlphaFactory): void;
+
+	/**
+	 * Fired after a new transaction is started.
+	 */
+	transactionStarted(): void;
```

Comment:
There's no test coverage that ensures these events are fired when appropriate. Could some of the tests from branch.spec.ts be ported over to treeCheckout.spec.ts?

Reply:
Are you okay waiting for a followup PR? I was planning on moving these to a separate transaction abstraction next, which can be unit tested.